src/Temperature-Accretech_UFseries/libs/accretech_uf.py

```diff
@@ -275,7 +279,19 @@ def acquire_status_byte(self, timeout: float = 10.0) -> int:
                 print("-->", now.strftime("%H:%M:%S"), "Function:", function_calling_name)
 
         # Wait for the SRQ event to occur
-        response = self.port.port.wait_on_event(self.event_type, int(timeout * 1000))  # conversion from s to ms
+        starttime = time.time()
+        while True: 
+            if time.time()-starttime < timeout:
+                try:
+                    # response = self.port.port.wait_on_event(self.event_type, int(timeout * 1000))  # conversion from s to ms
+                    response = self.port.port.wait_on_event(self.event_type, int(1000))  # waiting 1000 ms
+                    break
+                except pyvisa.errors.VisaIOError:
+                    # Timeout error
```

Comment:
Is VisaIOError always a timeout? What happens if it is something different than a timeout error?